src/main/java/uk/gov/ons/ctp/response/casesvc/endpoint/CaseEndpoint.java

```diff
@@ -284,15 +284,18 @@ private void createNewEventForAccessCodeAuthAttempt(final Case targetCase) throw
   }
 
   /**
-   * the GET endpoint to find case events by case id
+   * the GET endpoint to find case events, with optional timestamps by case id
    *
    * @param caseId to find by
-   * @return the case events found
+   * @param category type to collect case event timestamp
+   * @return the case events found, with optional timestamps (category required for this)
    * @throws CTPException something went wrong
    */
   @RequestMapping(value = "/{caseId}/events", method = RequestMethod.GET)
   public ResponseEntity<List<CaseEventDTO>> findCaseEventsByCaseId(
-      @PathVariable("caseId") final UUID caseId) throws CTPException {
+      @PathVariable("caseId") final UUID caseId,
+      @RequestParam(value = "category", required = false) final List<String> categories)
```

Comment:
actually, I was being dumb... we can use the enumeration type and Spring will make sure that only valid enum values are chosen and return a 400 bad request if garbage is sent